### PR TITLE
Fixes #6844 - updates /api/v2 view to display v2 resources

### PR DESCRIPTION
**Pull request by dustints**

### CODE REVIEW

File: app/views/api/v2/home/index.json.rabl [hunk 1]
```diff
@@ -2,10 +2,14 @@ object false
...
   # gather index methods of resources
-  index_method_description_apis = Apipie.app.resource_descriptions[Apipie.configuration.default_version].map do |name, resource_description|
-    if (description = Apipie.app["#{name}#index"])
-      description.method_apis_to_json.first
+  index_method_description_apis = Apipie.app.resource_descriptions['v2'].map do |name, resource_description|
+    method_description = resource_description.method_description(:index)
+    if(method_description)
+      method_description.method_apis_to_json.first
     end
+    #if (description = Apipie.app["#{name}#index"])
```
Comment (domcleal) [inline app/views/api/v2/home/index.json.rabl]: please remove commented out code

**Comment by domcleal** [comment]
Please review @isratrade, thanks.

**Comment by dustints** [comment]
updated, here's some of the sample output:

GET /api/v2

```
{
    "links": {
        "List all compute resources": "/api/compute_resources",
        "Get statistics": "/api/statistics",
        "List of override values for a specific smart variable": "/api/smart_variables/:smart_variable_id/override_values",
        "List all architectures": "/api/architectures",
        "List all installation media": "/api/media",
        "List all images for a compute resource": "/api/compute_resources/:compute_resource_id/images",
        "List tasks of given organization": "/api/organizations/:organization_id/tasks",
        "List all reports": "/api/reports",
        "List all smart variables": "/api/smart_variables",
        "List of realms": "/api/realms",
        "List all settings": "/api/settings",
        "Get dashboard details": "/api/dashboard",
        "List all partition tables": "/api/ptables",
        "List all Puppet classes": "/api/puppetclasses",
        "List all roles": "/api/roles",
        "List all filters": "/api/filters",
        "List installed plugins": "/api/plugins",
        "List all fact values": "/api/fact_values",
        "List all locations": "/api/locations",
        "List template combination": "/api/config_templates/:config_template_id/template_combinations",
        "List all template kinds": "/api/template_kinds",
        "List all bookmarks": "/api/bookmarks",
        "List all Puppet class IDs for host": "/api/hosts/:host_id/puppetclass_ids",
        "List all smart proxies": "/api/smart_proxies",
        "Show available API links": "/api",
        "List all global parameters": "/api/common_parameters",
        "List all host groups": "/api/hostgroups",
        "List all autosign entries": "/api/smart_proxies/smart_proxy_id/autosign",
        "List all interfaces for host": "/api/hosts/:host_id/interfaces",
        "List all operating systems": "/api/operatingsystems",
        "List of domains": "/api/domains",
        "List all audits": "/api/audits",
        "List of compute profiles": "/api/compute_profiles",
        "List all user groups": "/api/usergroups",
        "List all environments": "/api/environments",
        "List all permissions": "/api/permissions",
        "List all parameters for a host": "/api/hosts/:host_id/parameters",
        "List all LDAP authentication sources": "/api/auth_source_ldaps",
        "List all smart class parameters": "/api/smart_class_parameters",
        "List provisioning templates": "/api/config_templates",
        "List of subnets": "/api/subnets",
        "List default templates combinations for an operating system": "/api/operatingsystems/:operatingsystem_id/os_default_templates",
        "List all hosts": "/api/hosts",
        "List all hardware models": "/api/models",
        "List all Puppet class IDs for host group": "/api/hostgroups/:hostgroup_id/puppetclass_ids",
        "List of config groups": "/api/config_groups",
        "List all users": "/api/users",
        "List products": "/katello/api/products",
        "List filter rules": "/katello/api/content_view_filters/:content_view_filter_id/rules",
        "List packages": "/katello/api/repositories/:repository_id/packages",
        "List gpg keys": "/katello/api/gpg_keys",
        "Get list of errata associated with the host collection": "/katello/api/organizations/:organization_id/host_collections/:host_collection_id/errata",
        "List repository sets for a product": "/katello/api/products/:product_id/repository_sets",
        "List filters": "/katello/api/content_views/:content_view_id/filters",
        "List host collections": "/katello/api/host_collections",
        "Shows status of system and it's subcomponents": "/katello/api/ping",
        "List puppet modules": "/katello/api/puppet_modules",
        "List all organizations": "/katello/api/organizations",
        "List of enabled repositories": "/katello/api/repositories",
        "List sync plans": "/katello/api/organizations/:organization_id/sync_plans",
        "List distributions": "/katello/api/repositories/:repository_id/distributions",
        "List content hosts": "/katello/api/systems",
        "List organization subscriptions": "/katello/api/organizations/:organization_id/subscriptions",
        "List activation keys": "/katello/api/activation_keys",
        "List content views": "/katello/api/organizations/:organization_id/content_views",
        "List content view puppet modules": "/katello/api/content_views/:content_view_id/content_view_puppet_modules",
        "List content view versions": "/katello/api/content_view_versions",
        "List all capsules": "/katello/api/capsules",
        "List environments in an organization": "/katello/api/environments",
        "List errata": "/katello/api/errata",
        "Get status of repo synchronisation for given product": "/katello/api/organizations/:organization_id/products/:product_id/sync",
        "List package groups": "/katello/api/package_groups",
        "Show status": "/api/status"
    }
}
```

**Comment by domcleal** [comment]
Merged as 06bbfc3 for Foreman 1.6, thanks @dustint-rh.

**Comment by isratrade** [comment]
@domcleal, thanks @domcleal for merging this. I worked on this with @dustint-rh in Raliegh and was going to ACK it today.

**Pull request closed without merging**